src/request_helpers.js

```diff
@@ -0,0 +1,67 @@
+const { RetryableError, retryWithExpBackoff } = require('apify-shared/exponential_backoff');
+
+const GENERIC_UNHANDLED_ERROR_MESSAGE = 'Apify API Internal server error. Please report this issue to support@apify.com';
+
+/**
+ * Middleware includes the API token on all outbound requests.
+ * It runs runs before each request is sent out, allowing you to make tweaks to the request in a centralized spot.
+ */
+const includeApiToken = (request, z, bundle) => {
+    if (bundle.authData.token) {
+        request.params = request.params || {};
+        request.params.token = bundle.authData.token;
+    }
+    return request;
+};
+
+/**
+ * Middleware Parses nested data object into response.json
+ */
+const parseDataApiObject = (response) => {
+    response.json = response.json && response.json.data
+        ? response.json.data
+        : {};
+    return response;
+};
+
+/**
+ * This middleware log each bad response from Apify API.
+ */
+const validateApifyApiResponse = (response) => {
+    /**
+     * NOTE: If we throw RetryableError the request will be retried using exponential back off function
+     * in case we wrapped it in wrapRequestWithRetries function.
+     */
+    if (response.status >= 500) throw new RetryableError(GENERIC_UNHANDLED_ERROR_MESSAGE);
+    if (response.status === 429) throw new RetryableError('Apify API Rate Limit error. Try it again.');
+
+
+    if (response.status >= 300) {
+        let errorInfo;
+        try {
+            errorInfo = JSON.parse(response.content);
+        } catch (err) {
+            // This can be ignored
+        }
+        const errorMessage = errorInfo && errorInfo.error ? errorInfo.error.message : GENERIC_UNHANDLED_ERROR_MESSAGE;
+        throw new Error(errorMessage);
+    }
+
+    return response;
+};
+
+/**
+ * Wrapper for z.request() to use exponential back off calls
+ */
+const wrapRequestWithRetries = (request, options) => retryWithExpBackoff({
```

Comment:
I think that it would be cleaner to create function `zRequestExt(url, opts)` which would be `z.request()` wrapped by `retryWithExpBackoff()`  that you would use everywhere instead of wrapping `z.request()` everywhere again and again

Reply:
We need to wrap it because z.request is got from param in each zapier function.